File: app/src/main/java/io/mgba/Controller/PreferencesManager.java
package io.mgba.Controller;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

public class PreferencesManager {
    public static final String GAMES_DIRECTORY = "game_dir";
    private final Context mContext;
    private SharedPreferences.Editor editor;
    private SharedPreferences shared;

    public PreferencesManager(Context context) {
        this.mContext = context;
    }

    public void save(String key, String value) {
        SharedPreferences.Editor ed = getSharePreferencesEditor();

        ed.putString(key, value);
        ed.apply();
    }

    public String get(String key, String defaultValue){
        SharedPreferences prefs = getSharePreferences();
        return prefs.getString(key, defaultValue);
    }

    private SharedPreferences.Editor getSharePreferencesEditor() {
        if(editor == null){
            SharedPreferences sh = getSharePreferences();
            editor = shared.edit();
        }

        return editor;
    }

    private SharedPreferences getSharePreferences() {
        if(shared == null){
            shared = PreferenceManager.getDefaultSharedPreferences(mContext);
        }

        return shared;
    }
}

File: app/src/main/java/io/mgba/utilities/device/PreferencesManager.kt
package io.mgba.utilities.device

import android.annotation.SuppressLint
import android.content.Context.MODE_PRIVATE
import io.mgba.mgba

/**
 * Controller responsible with the interaction of the SharedPreferences.
 * It handles the saving/loading of the preferences.
 */
@SuppressLint("StaticFieldLeak")
object PreferencesManager {

    val GAMES_DIRECTORY = "game_dir"
    val SETUP_DONE = "setup_done"

    private val preferences = mgba.context.getSharedPreferences("MeBeSafe", MODE_PRIVATE)
    private val editor = preferences.edit()

    fun save(key: String, value: String) {
        editor.putString(key, value)
        editor.apply()
    }

    fun save(key: String, value: Boolean) {
        editor.putBoolean(key, value)
        editor.apply()
    }

    fun get(key: String, defaultValue: String): String = preferences.getString(key, defaultValue)!!
    fun get(key: String, defaultValue: Boolean): Boolean = preferences.getBoolean(key, defaultValue)
}
